### PR TITLE
Fix error with latest conda-build that caused every package to be rebuilt

### PR DESCRIPTION
**Pull request by rmcgibbo**

### CODE REVIEW

**Comment by rmcgibbo** [comment]
Fixes https://github.com/omnia-md/conda-recipes/pull/541#issuecomment-226576139

**Comment by rmcgibbo** [comment]
And https://github.com/omnia-md/conda-recipes/pull/540#issuecomment-226818137

**Comment by davidlmobley** [comment]
Ah, thanks. I was just going to ask if it was a bug with my recipe, or some other kind of bug... This clarifies it.

File: conda-build-all [hunk 1]
```diff
@@ -299,6 +300,10 @@ def required_builds(metadatas, pythons, numpys, channel_urls, verbose):
     index = conda.api.get_index(
         channel_urls=[url_path(conda_build_config.croot)] + list(channel_urls),
         prepend=False)
+    index_by_pkgname = defaultdict(lambda: {})
```
Comment (mpharrigan) [inline conda-build-all]: You can do `defaultdict(dict)`

**Comment by mpharrigan** [comment]
Thanks Robert!

**Comment by jchodera** [comment]
Thanks, everybody!

**Pull request merged**